### PR TITLE
[ally] Keyboard navigation in security passcode view

### PR DESCRIPTION
**Pull request by Aitorbp**
## Related Issues
App: https://github.com/owncloud/android/issues/4447

- [x] Add changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
- [x] Add feature to Release Notes in `ReleaseNotesViewModel.kt` creating a new `ReleaseNote()` with String resources (if required)

_____

### CODE REVIEW

**Comment by Aitorbp** [comment]
The toolbar is not accessible by the keyboard because of technical restriction. See this [comment](https://github.com/owncloud/android/issues/4366#issuecomment-2236411592) to more details.

File: owncloudApp/src/main/java/com/owncloud/android/presentation/security/passcode/PassCodeActivity.kt [hunk 1]
```diff
@@ -434,15 +444,33 @@ class PassCodeActivity : AppCompatActivity(), NumberKeyboardListener, EnableBiom
...
             }
         }
     }
 
+    override fun onKeyUp(keyCode: Int, event: KeyEvent?): Boolean {
+        return when (keyCode) {
+            KeyEvent.KEYCODE_TAB -> {
+                findViewById<View>(R.id.lock_time).requestFocus()
+                true
+            }
```
Comment (jesmrec) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/security/passcode/PassCodeActivity.kt:463]: Why don't you add here a blank line in between the `when` options? in the code above 👆  , you added blanks in every line of existing code

File: owncloudApp/src/main/java/com/owncloud/android/presentation/security/passcode/PassCodeActivity.kt [hunk 2]
```diff
@@ -434,15 +444,33 @@ class PassCodeActivity : AppCompatActivity(), NumberKeyboardListener, EnableBiom
...
+            KeyEvent.KEYCODE_DPAD_DOWN -> {
+                if (!findViewById<View>(R.id.numberKeyboard).hasFocus()) {
+                    findViewById<View>(R.id.numberKeyboard).requestFocus()
+                }
+                true
+            }
```
Comment (jesmrec) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/security/passcode/PassCodeActivity.kt:469]: Same here

**Review by jesmrec — CHANGES_REQUESTED** [review]

File: owncloudApp/src/main/java/com/owncloud/android/presentation/security/passcode/PassCodeActivity.kt (hunk 1 above)
Comment (Aitorbp) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/security/passcode/PassCodeActivity.kt:463]: Changed

**Comment by jesmrec** [comment]
Things i am missing here:

- Navigation through buttons is correct, but, numerical keyboard should work as well. User should be able to type/delete the passcode with the physical keyboard in the same way as clicking buttons. 

- Navigation through buttons is a keyboard trap in the settings. I did not find the way to jump from the buttons to the topbar, in case i want to cancel the view

- Are the focused buttons with the keyboard fulfilling contrast requirements?

**Comment by Aitorbp** [comment]
> Things i am missing here:
> 
> * Navigation through buttons is correct, but, numerical keyboard should work as well. User should be able to type/delete the passcode with the physical keyboard in the same way as clicking buttons.

Now It should work.

> * Navigation through buttons is a keyboard trap in the settings. I did not find the way to jump from the buttons to the topbar, in case i want to cancel the view

Try again. With my external keyboard the ESC key is working. But I added something else. Let´s see.
> * Are the focused buttons with the keyboard fulfilling contrast requirements? 

No, It is not meeting the contrast requirements
@jesmrec

**Comment by jesmrec** [comment]
After last commit:

- typing numbers with physical keyboard is now posible. Numbers are displayed while typing. It'd be better not to be shown but this is not a big problem

- `Esc` works to dismiss the view 💯

**Comment by jesmrec** [comment]
Let's move the contrast thing to the global topic, then...

this one is approved, at least for a first iteration.

**Pull request merged**